RollingUpdateTest.shouldPerformRollingUpgrade flaky

**Summary**

- How often does the test fail? more than once so far
- Does it block your work? no
- Do we suspect that it is a real failure? unknown

**Failures**

<details><summary>Example assertion failure</summary>
<pre>
org.awaitility.core.ConditionTimeoutException: Condition with alias 'broker is removed from topology' didn't complete within 10 seconds because assertion condition defined as a lambda expression in io.zeebe.test.RollingUpdateTest Expected topology not to contain broker with ID 0, but found the following: [[0, 1, 2]].
	at org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:165)
	at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:119)
	at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:31)
	at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:895)
	at org.awaitility.core.ConditionFactory.untilAsserted(ConditionFactory.java:679)
	at io.zeebe.test.RollingUpdateTest.shouldPerformRollingUpgrade(RollingUpdateTest.java:141)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeLazy(JUnitCoreWrapper.java:119)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:87)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:157)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
Caused by: java.lang.AssertionError: Expected topology not to contain broker with ID 0, but found the following: [[0, 1, 2]]
	at io.zeebe.test.RollingUpdateTest.assertTopologyDoesNotContainerBroker(RollingUpdateTest.java:219)
	at io.zeebe.test.RollingUpdateTest.lambda$shouldPerformRollingUpgrade$6(RollingUpdateTest.java:141)
	at org.awaitility.core.AssertionCondition.lambda$new$0(AssertionCondition.java:53)
	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:222)
	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:209)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
</pre>
</details>

**Hypotheses**


**Logs**


<details><summary>Logs</summary>
<pre>
22:41:39.567 [main] [] INFO  io.zeebe.test - Test started: shouldPerformRollingUpgrade(io.zeebe.test.RollingUpdateTest)
22:41:48.780 [testcontainers-lifecycle-2] [] INFO  org.testcontainers.dockerclient.EnvironmentAndSystemPropertyClientProviderStrategy - Found docker client settings from environment
22:41:48.781 [testcontainers-lifecycle-2] [] INFO  org.testcontainers.dockerclient.DockerClientProviderStrategy - Found Docker environment with Environment variables, system properties and defaults. Resolved dockerHost=tcp://localhost:2375
22:41:49.693 [testcontainers-lifecycle-2] [] INFO  org.testcontainers.DockerClientFactory - Docker host IP address is localhost
22:41:49.793 [testcontainers-lifecycle-2] [] INFO  org.testcontainers.DockerClientFactory - Connected to docker: 
  Server Version: 18.09.4
  API Version: 1.39
  Operating System: Alpine Linux v3.9 (containerized)
  Total Memory: 120871 MB
22:41:49.990 [testcontainers-lifecycle-2] [] WARN  org.testcontainers.utility.RegistryAuthLocator - Failure when attempting to lookup auth config (dockerImageName: testcontainersofficial/ryuk:0.3.0, configFile: /root/.docker/config.json. Falling back to docker-java default behaviour. Exception message: /root/.docker/config.json (No such file or directory)
22:41:51.285 [tc-okhttp-stream-1215724794] [] INFO  org.testcontainers.DockerClientFactory - Starting to pull image
22:41:51.379 [tc-okhttp-stream-1215724794] [] INFO  org.testcontainers.DockerClientFactory - Pulling image layers:  0 pending,  0 downloaded,  0 extracted, (0 bytes/0 bytes)
22:41:53.763 [testcontainers-lifecycle-2] [] INFO  org.testcontainers.DockerClientFactory - Ryuk started - will monitor and terminate Testcontainers containers on JVM exit
22:41:53.764 [testcontainers-lifecycle-2] [] INFO  org.testcontainers.DockerClientFactory - Checking the system...
22:41:53.765 [testcontainers-lifecycle-2] [] INFO  org.testcontainers.DockerClientFactory - ✔︎ Docker server version should be at least 1.6.0
22:41:54.094 [testcontainers-lifecycle-2] [] INFO  org.testcontainers.DockerClientFactory - ✔︎ Docker environment should have more than 2GB free disk space
22:41:54.177 [testcontainers-lifecycle-2] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling docker image: camunda/zeebe:0.23.0. Please be patient; this may take some time but only needs to be done once.
22:41:54.177 [testcontainers-lifecycle-1] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling docker image: camunda/zeebe:0.23.0. Please be patient; this may take some time but only needs to be done once.
22:41:54.178 [testcontainers-lifecycle-0] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling docker image: camunda/zeebe:0.23.0. Please be patient; this may take some time but only needs to be done once.
22:41:55.391 [tc-okhttp-stream-576289301] [] INFO  🐳 [camunda/zeebe:0.23.0] - Starting to pull image
22:41:55.393 [tc-okhttp-stream-576289301] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending,  0 downloaded,  0 extracted, (0 bytes/0 bytes)
22:41:55.464 [tc-okhttp-stream-2069050773] [] INFO  🐳 [camunda/zeebe:0.23.0] - Starting to pull image
22:41:55.465 [tc-okhttp-stream-2069050773] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending,  0 downloaded,  0 extracted, (0 bytes/0 bytes)
22:41:55.466 [tc-okhttp-stream-270939684] [] INFO  🐳 [camunda/zeebe:0.23.0] - Starting to pull image
22:41:55.467 [tc-okhttp-stream-270939684] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending,  0 downloaded,  0 extracted, (0 bytes/0 bytes)
22:41:55.966 [tc-okhttp-stream-270939684] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers: 11 pending,  1 downloaded,  0 extracted, (8 MB/? MB)
22:41:55.968 [tc-okhttp-stream-270939684] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers: 10 pending,  2 downloaded,  0 extracted, (16 MB/? MB)
22:41:55.969 [tc-okhttp-stream-2069050773] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers: 11 pending,  1 downloaded,  0 extracted, (8 MB/? MB)
22:41:55.970 [tc-okhttp-stream-576289301] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers: 11 pending,  1 downloaded,  0 extracted, (8 MB/? MB)
22:41:55.970 [tc-okhttp-stream-2069050773] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers: 10 pending,  2 downloaded,  0 extracted, (16 MB/? MB)
22:41:55.971 [tc-okhttp-stream-576289301] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers: 10 pending,  2 downloaded,  0 extracted, (16 MB/? MB)
22:41:56.364 [tc-okhttp-stream-2069050773] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  9 pending,  3 downloaded,  0 extracted, (46 MB/? MB)
22:41:56.364 [tc-okhttp-stream-576289301] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  9 pending,  3 downloaded,  0 extracted, (46 MB/? MB)
22:41:56.367 [tc-okhttp-stream-270939684] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  9 pending,  3 downloaded,  0 extracted, (46 MB/? MB)
22:41:56.464 [tc-okhttp-stream-2069050773] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  8 pending,  4 downloaded,  0 extracted, (46 MB/? MB)
22:41:56.467 [tc-okhttp-stream-270939684] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  8 pending,  4 downloaded,  0 extracted, (46 MB/? MB)
22:41:56.464 [tc-okhttp-stream-576289301] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  8 pending,  4 downloaded,  0 extracted, (46 MB/? MB)
22:41:56.564 [tc-okhttp-stream-576289301] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  7 pending,  5 downloaded,  0 extracted, (46 MB/? MB)
22:41:56.568 [tc-okhttp-stream-270939684] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  7 pending,  5 downloaded,  0 extracted, (46 MB/? MB)
22:41:56.569 [tc-okhttp-stream-2069050773] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  7 pending,  5 downloaded,  0 extracted, (46 MB/? MB)
22:41:56.873 [tc-okhttp-stream-270939684] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  6 pending,  6 downloaded,  0 extracted, (46 MB/? MB)
22:41:56.874 [tc-okhttp-stream-2069050773] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  6 pending,  6 downloaded,  0 extracted, (46 MB/? MB)
22:41:56.875 [tc-okhttp-stream-576289301] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  6 pending,  6 downloaded,  0 extracted, (46 MB/? MB)
22:41:56.978 [tc-okhttp-stream-2069050773] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  5 pending,  7 downloaded,  0 extracted, (56 MB/? MB)
22:41:56.978 [tc-okhttp-stream-270939684] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  5 pending,  7 downloaded,  0 extracted, (56 MB/? MB)
22:41:56.981 [tc-okhttp-stream-576289301] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  5 pending,  7 downloaded,  0 extracted, (56 MB/? MB)
22:41:57.365 [tc-okhttp-stream-270939684] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  4 pending,  8 downloaded,  0 extracted, (84 MB/? MB)
22:41:57.367 [tc-okhttp-stream-2069050773] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  4 pending,  8 downloaded,  0 extracted, (84 MB/? MB)
22:41:57.369 [tc-okhttp-stream-576289301] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  4 pending,  8 downloaded,  0 extracted, (84 MB/? MB)
22:41:57.774 [tc-okhttp-stream-576289301] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  3 pending,  9 downloaded,  0 extracted, (119 MB/? MB)
22:41:57.774 [tc-okhttp-stream-2069050773] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  3 pending,  9 downloaded,  0 extracted, (119 MB/? MB)
22:41:57.775 [tc-okhttp-stream-270939684] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  3 pending,  9 downloaded,  0 extracted, (119 MB/? MB)
22:41:58.274 [tc-okhttp-stream-576289301] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  2 pending, 10 downloaded,  0 extracted, (196 MB/? MB)
22:41:58.276 [tc-okhttp-stream-2069050773] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  2 pending, 10 downloaded,  0 extracted, (196 MB/? MB)
22:41:58.276 [tc-okhttp-stream-270939684] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  2 pending, 10 downloaded,  0 extracted, (196 MB/? MB)
22:41:58.764 [tc-okhttp-stream-270939684] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  1 pending, 11 downloaded,  0 extracted, (277 MB/? MB)
22:41:58.765 [tc-okhttp-stream-2069050773] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  1 pending, 11 downloaded,  0 extracted, (277 MB/? MB)
22:41:58.766 [tc-okhttp-stream-576289301] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  1 pending, 11 downloaded,  0 extracted, (277 MB/? MB)
22:41:58.864 [tc-okhttp-stream-270939684] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  0 extracted, (285 MB/306 MB)
22:41:58.868 [tc-okhttp-stream-2069050773] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  0 extracted, (285 MB/306 MB)
22:41:58.864 [tc-okhttp-stream-576289301] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  0 extracted, (285 MB/306 MB)
22:41:58.995 [tc-okhttp-stream-576289301] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  1 extracted, (285 MB/306 MB)
22:41:58.995 [tc-okhttp-stream-2069050773] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  1 extracted, (285 MB/306 MB)
22:41:58.997 [tc-okhttp-stream-270939684] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  1 extracted, (285 MB/306 MB)
22:41:59.686 [tc-okhttp-stream-576289301] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  2 extracted, (293 MB/306 MB)
22:41:59.687 [tc-okhttp-stream-270939684] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  2 extracted, (293 MB/306 MB)
22:41:59.689 [tc-okhttp-stream-2069050773] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  2 extracted, (293 MB/306 MB)
22:42:00.066 [tc-okhttp-stream-270939684] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  3 extracted, (295 MB/306 MB)
22:42:00.066 [tc-okhttp-stream-2069050773] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  3 extracted, (295 MB/306 MB)
22:42:00.072 [tc-okhttp-stream-576289301] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  3 extracted, (295 MB/306 MB)
22:42:00.464 [tc-okhttp-stream-2069050773] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  4 extracted, (300 MB/306 MB)
22:42:00.465 [tc-okhttp-stream-576289301] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  4 extracted, (300 MB/306 MB)
22:42:00.466 [tc-okhttp-stream-270939684] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  4 extracted, (300 MB/306 MB)
22:42:00.684 [tc-okhttp-stream-2069050773] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  5 extracted, (300 MB/306 MB)
22:42:00.686 [tc-okhttp-stream-270939684] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  5 extracted, (300 MB/306 MB)
22:42:00.686 [tc-okhttp-stream-576289301] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  5 extracted, (300 MB/306 MB)
22:42:04.005 [tc-okhttp-stream-2069050773] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  6 extracted, (302 MB/306 MB)
22:42:04.005 [tc-okhttp-stream-270939684] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  6 extracted, (302 MB/306 MB)
22:42:04.005 [tc-okhttp-stream-576289301] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  6 extracted, (302 MB/306 MB)
22:42:04.288 [tc-okhttp-stream-2069050773] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  7 extracted, (302 MB/306 MB)
22:42:04.288 [tc-okhttp-stream-270939684] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  7 extracted, (302 MB/306 MB)
22:42:04.289 [tc-okhttp-stream-576289301] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  7 extracted, (302 MB/306 MB)
22:42:04.590 [tc-okhttp-stream-270939684] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  8 extracted, (302 MB/306 MB)
22:42:04.590 [tc-okhttp-stream-2069050773] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  8 extracted, (302 MB/306 MB)
22:42:04.590 [tc-okhttp-stream-576289301] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  8 extracted, (302 MB/306 MB)
22:42:07.764 [tc-okhttp-stream-576289301] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  9 extracted, (303 MB/306 MB)
22:42:07.764 [tc-okhttp-stream-270939684] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  9 extracted, (303 MB/306 MB)
22:42:07.764 [tc-okhttp-stream-2069050773] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded,  9 extracted, (303 MB/306 MB)
22:42:10.982 [tc-okhttp-stream-576289301] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded, 10 extracted, (305 MB/306 MB)
22:42:10.982 [tc-okhttp-stream-270939684] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded, 10 extracted, (305 MB/306 MB)
22:42:10.985 [tc-okhttp-stream-2069050773] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded, 10 extracted, (305 MB/306 MB)
22:42:11.107 [tc-okhttp-stream-576289301] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded, 11 extracted, (305 MB/306 MB)
22:42:11.164 [tc-okhttp-stream-270939684] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded, 11 extracted, (305 MB/306 MB)
22:42:11.166 [tc-okhttp-stream-2069050773] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded, 11 extracted, (305 MB/306 MB)
22:42:11.288 [tc-okhttp-stream-2069050773] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded, 12 extracted, (306 MB/306 MB)
22:42:11.288 [tc-okhttp-stream-270939684] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded, 12 extracted, (306 MB/306 MB)
22:42:11.295 [tc-okhttp-stream-576289301] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pulling image layers:  0 pending, 12 downloaded, 12 extracted, (306 MB/306 MB)
22:42:11.383 [tc-okhttp-stream-576289301] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pull complete. 12 layers, pulled in 15s (downloaded 306 MB at 20 MB/s)
22:42:11.383 [tc-okhttp-stream-2069050773] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pull complete. 12 layers, pulled in 15s (downloaded 306 MB at 20 MB/s)
22:42:11.391 [tc-okhttp-stream-270939684] [] INFO  🐳 [camunda/zeebe:0.23.0] - Pull complete. 12 layers, pulled in 15s (downloaded 306 MB at 20 MB/s)
22:42:11.488 [testcontainers-lifecycle-0] [] INFO  🐳 [camunda/zeebe:0.23.0] - Creating container for image: camunda/zeebe:0.23.0
22:42:11.488 [testcontainers-lifecycle-2] [] INFO  🐳 [camunda/zeebe:0.23.0] - Creating container for image: camunda/zeebe:0.23.0
22:42:11.497 [testcontainers-lifecycle-1] [] INFO  🐳 [camunda/zeebe:0.23.0] - Creating container for image: camunda/zeebe:0.23.0
22:42:27.181 [testcontainers-lifecycle-1] [] INFO  🐳 [camunda/zeebe:0.23.0] - Starting container with ID: b3b1d4bfa04d295d544c97e0ab953d2352fd6e8430b2673cbfff84fce5ec8850
22:42:27.388 [testcontainers-lifecycle-0] [] INFO  🐳 [camunda/zeebe:0.23.0] - Starting container with ID: a45e071e10fd7ef8de6a892d333326037597e0c287c48a2a66d8757a4cad40e0
22:42:29.197 [testcontainers-lifecycle-2] [] INFO  🐳 [camunda/zeebe:0.23.0] - Starting container with ID: de5ea7f5c50e408b49dd4aed0be0986c73df4ebc891c3581f8248a1e3e5075b5
22:42:29.964 [testcontainers-lifecycle-1] [] INFO  🐳 [camunda/zeebe:0.23.0] - Container camunda/zeebe:0.23.0 is starting: b3b1d4bfa04d295d544c97e0ab953d2352fd6e8430b2673cbfff84fce5ec8850
22:42:29.999 [testcontainers-lifecycle-1] [] INFO  org.testcontainers.containers.wait.strategy.HttpWaitStrategy - /zeebe-broker-756U1o-dwADau: Waiting for 60 seconds for URL: http://localhost:32783/ready
22:42:30.182 [testcontainers-lifecycle-0] [] INFO  🐳 [camunda/zeebe:0.23.0] - Container camunda/zeebe:0.23.0 is starting: a45e071e10fd7ef8de6a892d333326037597e0c287c48a2a66d8757a4cad40e0
22:42:30.299 [testcontainers-lifecycle-2] [] INFO  🐳 [camunda/zeebe:0.23.0] - Container camunda/zeebe:0.23.0 is starting: de5ea7f5c50e408b49dd4aed0be0986c73df4ebc891c3581f8248a1e3e5075b5
22:42:31.002 [testcontainers-lifecycle-2] [] INFO  org.testcontainers.containers.wait.strategy.HttpWaitStrategy - /zeebe-broker-kDQYop-4SMrqR: Waiting for 60 seconds for URL: http://localhost:32791/ready
22:42:31.002 [testcontainers-lifecycle-0] [] INFO  org.testcontainers.containers.wait.strategy.HttpWaitStrategy - /zeebe-broker-ScYUHQ-6PCr5S: Waiting for 60 seconds for URL: http://localhost:32787/ready
22:43:01.066 [testcontainers-lifecycle-1] [] INFO  🐳 [camunda/zeebe:0.23.0] - Container camunda/zeebe:0.23.0 started in PT1M16.773725S
22:43:01.177 [testcontainers-lifecycle-2] [] INFO  🐳 [camunda/zeebe:0.23.0] - Container camunda/zeebe:0.23.0 started in PT1M16.88754S
22:43:07.293 [testcontainers-lifecycle-0] [] INFO  🐳 [camunda/zeebe:0.23.0] - Container camunda/zeebe:0.23.0 started in PT1M23.00284S
22:43:21.304 [main] [] INFO  🐳 [camunda/zeebe:current-test] - Creating container for image: camunda/zeebe:current-test
22:43:21.414 [main] [] INFO  🐳 [camunda/zeebe:current-test] - Starting container with ID: 53344bc0f8ed80f79163a8564d7bd0e2171dc847457a2292b5c9904f844a5db5
22:43:21.991 [main] [] INFO  🐳 [camunda/zeebe:current-test] - Container camunda/zeebe:current-test is starting: 53344bc0f8ed80f79163a8564d7bd0e2171dc847457a2292b5c9904f844a5db5
22:43:22.002 [main] [] INFO  org.testcontainers.containers.wait.strategy.HttpWaitStrategy - /zeebe-broker-kGKYbm-bcf4Lg: Waiting for 60 seconds for URL: http://localhost:32807/ready
22:43:40.380 [main] [] INFO  🐳 [camunda/zeebe:current-test] - Container camunda/zeebe:current-test started in PT19.076607S
22:43:50.223 [main] [] INFO  🐳 [camunda/zeebe:current-test] - Creating container for image: camunda/zeebe:current-test
22:43:50.375 [main] [] INFO  🐳 [camunda/zeebe:current-test] - Starting container with ID: 92f64bb43950d6e4cc99062d456e3ac83eda42c691856a70a50550cec664485d
22:43:51.803 [main] [] INFO  🐳 [camunda/zeebe:current-test] - Container camunda/zeebe:current-test is starting: 92f64bb43950d6e4cc99062d456e3ac83eda42c691856a70a50550cec664485d
22:43:51.812 [main] [] INFO  org.testcontainers.containers.wait.strategy.HttpWaitStrategy - /zeebe-broker-Kb7N8P-As7LfT: Waiting for 60 seconds for URL: http://localhost:32821/ready
22:44:15.083 [main] [] INFO  🐳 [camunda/zeebe:current-test] - Container camunda/zeebe:current-test started in PT24.861395S
</pre>
</details>

--- a/upgrade-tests/src/test/java/io/zeebe/test/RollingUpdateTest.java
+++ b/upgrade-tests/src/test/java/io/zeebe/test/RollingUpdateTest.java
@@ -19,13 +19,13 @@ import io.zeebe.model.bpmn.BpmnModelInstance;
 import io.zeebe.test.util.asserts.TopologyAssert;
 import io.zeebe.util.VersionUtil;
 import java.time.Duration;
-import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
 import java.util.Optional;
+import java.util.concurrent.CopyOnWriteArrayList;
 import java.util.concurrent.TimeUnit;
 import java.util.stream.Collectors;
 import java.util.stream.IntStream;
@@ -133,10 +133,10 @@ public class RollingUpdateTest {
 
         container.shutdownGracefully(Duration.ofSeconds(30));
 
-        // until previous version points to 0.24, we cannot yet tune failure detection to be very
-        // fast, so
+        // until previous version points to 0.24, we cannot yet tune failure detection to be fast,
+        // so wait long enough for the broker to be removed even in slower systems
         Awaitility.await("broker is removed from topology")
-            .atMost(Duration.ofSeconds(10))
+            .atMost(Duration.ofSeconds(20))
             .pollInterval(Duration.ofMillis(100))
             .untilAsserted(() -> assertTopologyDoesNotContainerBroker(client, brokerId));
 
@@ -160,7 +160,8 @@ public class RollingUpdateTest {
           activatedJobs.compute(
               job.getWorkflowInstanceKey(),
               (ignored, list) -> {
-                final var appendedList = Optional.ofNullable(list).orElse(new ArrayList<>());
+                final var appendedList =
+                    Optional.ofNullable(list).orElse(new CopyOnWriteArrayList<>());
                 appendedList.add(job.getType());
                 return appendedList;
               });
@@ -254,7 +255,7 @@ public class RollingUpdateTest {
         .withEnv("ZEEBE_BROKER_CLUSTER_INITIALCONTACTPOINTS", initialContactPoints)
         .withEnv("ZEEBE_BROKER_CLUSTER_GOSSIPFAILURETIMEOUT", "5000")
         .withEnv("ZEEBE_BROKER_CLUSTER_GOSSIPINTERVAL", "100")
-        .withEnv("ZEEBE_BROKER_CLUSTER_GOSSIPPROBEINTERVAL", "250")
+        .withEnv("ZEEBE_BROKER_CLUSTER_GOSSIPPROBEINTERVAL", "100")
         .withLogLevel(Level.DEBUG)
         .withDebug(false);
   }

